src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java

```diff
@@ -265,45 +262,40 @@ public void widerRequiredContext() {
         });
     }
 
-    @Test public void cleanupAfterRestart() {
+    @Test public void cleanupAfterRestart() throws Throwable {
         final String secret = "s3cr3t";
-        story.addStep(new Statement() {
-            @Override public void evaluate() throws Throwable {
-                FileCredentialsImpl c = new FileCredentialsImpl(CredentialsScope.GLOBAL, "creds", "sample", "secret.txt", SecretBytes.fromBytes(secret.getBytes()));
-                CredentialsProvider.lookupStores(story.j.jenkins).iterator().next().addCredentials(Domain.global(), c);
-                // TODO JENKINS-26398: story.j.createSlave("myslave", null, null) does not work since the slave root is deleted after restart.
-                story.j.jenkins.addNode(new DumbSlave("myslave", "", tmp.newFolder().getAbsolutePath(), "1", Node.Mode.NORMAL, "", story.j.createComputerLauncher(null), RetentionStrategy.NOOP, Collections.emptyList()));
-                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
-                p.setDefinition(new CpsFlowDefinition(""
-                        + "node('myslave') {"
-                        + "  withCredentials([file(variable: 'SECRET', credentialsId: 'creds')]) {\n"
-                        + "    semaphore 'cleanupAfterRestart'\n"
-                        + "    if (isUnix()) {sh 'cp $SECRET key'} else {bat 'copy %SECRET% key'}\n"
-                        + "  }\n"
-                        + "}", true));
-                WorkflowRun b = p.scheduleBuild2(0).waitForStart();
-                SemaphoreStep.waitForStart("cleanupAfterRestart/1", b);
-            }
+        rr.then(r -> {
+            FileCredentialsImpl c = new FileCredentialsImpl(CredentialsScope.GLOBAL, "creds", "sample", "secret.txt", SecretBytes.fromBytes(secret.getBytes()));
+            CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), c);
+            r.createSlave("myslave", null, null);
+            WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+            p.setDefinition(new CpsFlowDefinition(""
+                    + "node('myslave') {"
+                    + "  withCredentials([file(variable: 'SECRET', credentialsId: 'creds')]) {\n"
+                    + "    semaphore 'cleanupAfterRestart'\n"
+                    + "    if (isUnix()) {sh 'cp \"$SECRET\" key'} else {bat 'copy %SECRET% key'}\n"
```

Comment:
Needed to quote the secret path; since it is now inside `$JENKINS_HOME` it will contain the `j h` subsequence.